benchmarks/dataset_read_benchmark.py

```diff
@@ -27,34 +27,60 @@ def run(self, source, case=None, **kwargs):
         cases = self.get_cases(case, kwargs)
         for source in self.get_sources(source):
             tags = self.get_tags(kwargs, source)
-            schema = self._get_schema(source)
+            format_str = self._get_format_str(source)
+            schema = self._get_schema(source, format_str)
             s3 = pyarrow.fs.S3FileSystem(region=source.region)
             for case in cases:
                 (pre_buffer,) = case
-                legacy, parquet = self._get_format(pre_buffer)
+                legacy, format_ = self._get_format(pre_buffer, format_str)
                 case = (None,) if legacy else case
                 dataset = pyarrow.dataset.FileSystemDataset.from_paths(
                     source.paths,
                     schema=schema,
-                    format=parquet,
+                    format=format_,
                     filesystem=s3,
                 )
-                f = self._get_benchmark_function(dataset)
+                supports_async = self._get_supports_async(dataset, format_str)
+                tags["async"] = supports_async
```

Comment:
This new tag will be applied to the existing dataset-read benchmark cases, resulting in a new benchmark cases which means we won't be able to compare them to historic results.

Instead, you might want to not tag when supports_async is false, and only tag when supports_async is true.

For example:

The existing case is: `pre_buffer=false, nyctaxi_multi_parquet_s3`

Will now become: `pre_buffer=false, async=false, nyctaxi_multi_parquet_s3`

So we won't be able to compare them.

Instead consider keeping it `pre_buffer=false, nyctaxi_multi_parquet_s3` when `async=false`.

The alternative is a data migration which I can help with.

Let me know if that wasn't clear -- it was a bit hard to try and explain.

Reply:
The explanation was very clear :).  I was wondering about it a bit yesterday too and decided this way pretty much on a coin flip and not really understanding the implication of updating tags.  I've done as you suggested and will only add the tag if it is True.